k8_cortx_cloud/deploy-cortx-cloud.sh

```diff
@@ -883,11 +886,55 @@ function deployCortxSecrets()
     done
 }
 
+function silentKill()
+{
+    kill $1
+    wait $1 2> /dev/null
+}
+
+function waitForAllDeploymentsAvailable()
+{
+    INIT_DELAY=$1
+    shift
+    NEXT_DELAY=$1
+    shift
+    DEPL_STR=$1
+    shift
+
+    START=$SECONDS
+    (while true; do sleep 1; echo -n "."; done)&
+    DOTPID=$!
+    trap "silentKill $DOTPID" 0
+    
+    # Initial wait
+    FAIL=0
+    kubectl wait --for=condition=available --timeout=$INIT_DELAY $@
```

Comment:
@osowski and I discussed this "double-check" idea (if I understood correctly).  But now that I look at it, I'm not sure it buys us anything.  I would consider removing the "secondary wait" and just having a single, simple timeout.

Reply:
We want to be somewhat resilient in how we handle failures. Doing a simple second check helps in a "micorservices" oriented way... let's give it a reasonable timeout (calculated based on deployment size). If it fails, try it again with the same timeout (as it will either be successful in less than that timeout if our timeout formula is accurate or it will fail again because it's always going to fail). Then the deployment will succeed if it just needs more time, but it will also fail if it's always going to fail.